emloop/tests/cli/common_test.py

```diff
@@ -343,3 +355,20 @@ def test_config_file_is_incorrect(tmpdir, caplog):
     config['main_loop']['on_incorrect_config'] = 'ignore'
     run(config=config, output_root=tmpdir)
     assert 'Extra arguments: {\'non-existent\': \'none\'}' not in caplog.text
+
+
+def test_run_with_eval_stream(tmpdir, caplog):
+    """Test that eval is called with given stream and not with EL_DEFAULT_TRAIN_STREAM."""
+    caplog.set_level(logging.INFO)
+
+    config = {'dataset': {'class': 'emloop.tests.cli.common_test.DummyEvalDataset'},
+              'stream': {'train': {'rotate': 20}},
+              'hooks': [{'LogProfile': {'epochs': 1}}, {'StopAfter': {'epochs': 1}}],
+              'model': {'class': 'emloop.tests.cli.common_test.DummyModel', 'io': {'in': ['a'], 'out': ['dummy']}}}
+
+    run(config=config, output_root=tmpdir, eval='valid')
+    
+    assert f'Running the evaluation of stream `{EL_DEFAULT_TRAIN_STREAM}`' not in caplog.text
+    assert f'T {EL_DEFAULT_TRAIN_STREAM}:' not in caplog.text
+    assert 'Running the evaluation of stream `valid`' in caplog.text
+    assert 'T valid:' in caplog.text
```

Comment:
I would prefer a different way. It is kind of dirty to check the log text for its contents. Can you add a dummy hook with `after_epoch_profile` method, that simply asserts that the provided `streams` parameter is `['eval']` ?